internal/service/eks/find.go

```diff
@@ -68,6 +68,56 @@ func FindAddonUpdateByClusterNameAddonNameAndID(ctx context.Context, conn *eks.E
 	return output.Update, nil
 }
 
+func FindAddonVersionByAddonNameAndKubernetesVersion(ctx context.Context, conn *eks.EKS, addonName, kubernetesVersion string, latest bool) (*eks.AddonVersionInfo, error) {
+	input := &eks.DescribeAddonVersionsInput{
+		AddonName:         aws.String(addonName),
+		KubernetesVersion: aws.String(kubernetesVersion),
+	}
+	var version *eks.AddonVersionInfo
+
+	err := conn.DescribeAddonVersionsPagesWithContext(ctx, input, func(page *eks.DescribeAddonVersionsOutput, lastPage bool) bool {
+		if page == nil || len(page.Addons) == 0 {
+			return !lastPage
+		}
+
+		for _, addon := range page.Addons {
+			for i, addonVersion := range addon.AddonVersions {
+				if latest && i == 0 {
```

Comment:
Yes the API seems to guarantee that the first item will always be the latest

```
$ aws eks describe-addon-versions --addon-name coredns --kubernetes-version 1.20 | jq .addons[].addonVersions[].addonVersion --raw-output
v1.8.3-eksbuild.1
v1.8.0-eksbuild.1

$ aws eks describe-addon-versions --addon-name kube-proxy --kubernetes-version 1.20 | jq .addons[].addonVersions[].addonVersion --raw-output
v1.20.7-eksbuild.1
v1.20.4-eksbuild.2
v1.19.8-eksbuild.1
v1.19.6-eksbuild.2
v1.18.8-eksbuild.1

$ aws eks describe-addon-versions --addon-name vpc-cni --kubernetes-version 1.20 | jq .addons[].addonVersions[].addonVersion --raw-output
v1.10.2-eksbuild.1
v1.10.1-eksbuild.1
v1.9.3-eksbuild.1
v1.9.1-eksbuild.1
v1.9.0-eksbuild.1
v1.8.0-eksbuild.1
v1.7.10-eksbuild.1
v1.7.9-eksbuild.2
v1.7.6-eksbuild.2
v1.7.5-eksbuild.2
v1.6.3-eksbuild.2
```

However I'll check in with the service team to make sure of that.

Reply:
@AdamTylerLynch  I heard back from the service team and here's the feedback:
* **About latest version**: The version list is not guaranteed to be in order; it is a coincidence/implementation detail based on current backing data structure. Since it is an impl detail, it may change in future as it is not part of API contract.
* **About singularity of default version**: Correct, only one default version per clusterVersion. Guaranteed as it is part of API contract.

I'll rework the most_recent=true scenario to get all versions, sort them and return the lexicographically largest one.